### PR TITLE
[PIP-45] Add support for etcd to implement interfaces

### PR DESCRIPTION
**Pull request by complone**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

The subject of this issue is based on the storage of PIP-45 metadata, adding etcd as the registry, currently adding the interface layer, and the basic parameters that jetcd needs to configure. I will finish the work as soon as possible

[PIP-45](https://github.com/apache/pulsar/wiki/PIP-45%3A-Pluggable-metadata-interface)

*Explain here the context, and why you're making that change. What is the problem you're trying to solve.*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below and label this PR (if you have committer privilege).

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [ ] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: pulsar-metadata/src/main/java/org/apache/pulsar/metadata/api/MetadataStoreConfig.java [hunk 1]
```diff
@@ -39,4 +39,13 @@
      */
     @Builder.Default
     private final boolean allowReadOnlyOperations = false;
+
+    //etcd每次消息落盘的大小
```
Comment (Shoothzj) [inline pulsar-metadata/src/main/java/org/apache/pulsar/metadata/api/MetadataStoreConfig.java:43]: javadoc style may be better

File: pulsar-metadata/src/main/java/org/apache/pulsar/metadata/api/MetadataStoreConfig.java [hunk 2]
```diff
@@ -39,4 +39,13 @@
...
+    private final Integer maxInboundMessageSize;
+
+    //与注册中心链接的时间间隔
```
Comment (Shoothzj) [inline pulsar-metadata/src/main/java/org/apache/pulsar/metadata/api/MetadataStoreConfig.java:46]: I think we should use english here

File: pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/EtcdMetadataStore.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                .build();
+
+        this.kvClient = client.getKVClient();
+        this.leaseClient = client.getLeaseClient();
+
+
+    }
+
+    @Override
+    public CompletableFuture<Optional<GetResult>> get(String path) {
```
Comment (Jason918) [inline pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/EtcdMetadataStore.java:47]: WIP ?
There is no implementation.

Comment (complone) [reply pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/EtcdMetadataStore.java:47]: enmm...Yes, it is WIP

**Pull request closed without merging**